### PR TITLE
S3Opener init (develop)

### PR DESCRIPTION
**Pull request by nettorta**
and once again to appropriate branch

aws s3 opener

### CODE REVIEW

File: netort/resource.py [hunk 1]
```diff
@@ -403,4 +418,135 @@ def readline(self):
...
+            "port": 7480,
+            "is_secure": false
+        }
+    """
+
+    def __init__(self, uri, credentials_path='/etc/yandex-tank/s3credentials.json'):
+        # read s3 credentials
+        # FIXME move to default config? which section and how securely store the keys?
+        with open(credentials_path) as fname:
+            s3_credentials = json.loads(fname.read())
```
Comment (direvius) [inline netort/resource.py]: why not yaml? Yaml loader is backward compatible with JSON and will load JSON files also.

File: netort/resource.py [hunk 2]
```diff
@@ -403,4 +418,135 @@ def readline(self):
...
+            "aws_secret_access_key": "secret-id",
+            "host": "hostname.tld",
+            "port": 7480,
+            "is_secure": false
+        }
+    """
+
+    def __init__(self, uri, credentials_path='/etc/yandex-tank/s3credentials.json'):
+        # read s3 credentials
+        # FIXME move to default config? which section and how securely store the keys?
```
Comment (direvius) [inline netort/resource.py]: No, it's ok to store the credentials in external file.

File: netort/resource.py [hunk 3]
```diff
@@ -403,4 +418,135 @@ def readline(self):
...
+        self.uri = uri
+        urlparsed = urlparse(self.uri)
+        self.bucket_key = urlparsed.netloc
+        self.object_key = urlparsed.path.strip('/')
+        self._filename = None
+        self.conn = None
+
+    def connect(self):
+        if not boto:
+            raise RuntimeError('Install `boto` please')
```
Comment (direvius) [inline netort/resource.py]:
```suggestion
            raise RuntimeError("Install 'boto' python package manually please")
```

File: netort/resource.py [hunk 4]
```diff
@@ -403,4 +418,135 @@ def readline(self):
...
+            self._filename = self.get_file()
+        return self._filename
+
+    def tmpfile_path(self):
+        hasher = hashlib.md5()
+        hasher.update(self.hash)
+        return "/tmp/s3_%s.downloaded_resource" % hasher.hexdigest()
+
+    def upload_file(self, bucket_key, object_key, filepath):
+        # FIXME experimental feature! use at your own risk
```
Comment (direvius) [inline netort/resource.py]: Scary. Very scary.

File: netort/resource.py (hunk 2 above)
Comment (nettorta) [reply netort/resource.py]: So, where do you suggest to store auth sensitive information? Config definitely is not the right place. And, overload plugin stores token just like that.

Comment (direvius) [reply netort/resource.py]: I suggest external file =)

Comment (nettorta) [reply netort/resource.py]: troll.jpg

**Pull request merged**